src/ErrorSparseTask.cpp

```diff
@@ -208,16 +205,15 @@ void ErrorSparseTask::run(const Configuration& config) {
                   << " time(us): " << get_time_us()
                   << " time from start (sec): " << last_time << std::endl;
       } else if (config.get_model_type() == Configuration::COLLABORATIVE_FILTERING) {
-        last_error = std::sqrt(total_loss / total_num_features);
+        last_error = std::sqrt(total_loss / total_num_samples);
```

Comment:
I think we discussed this before. The collaborative filtering loss is the RMSE of each value in the matrix, so I think this should be ```total_num_features```.

Reply:
Reverted